### PR TITLE
feat: `custom-playback-speed` patch

### PR DESCRIPTION
**Pull request by bogadana**

### CODE REVIEW

**Comment by TheJeterLP** [comment]
Just a note, this will break when the integrations get merged

File: src/main/kotlin/app/revanced/patches/youtube/misc/customplaybackspeed/patch/CustomPlaybackSpeedPatch.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        arrayGenMethodImpl.replaceInstruction(sizeCallIndex + 1, 
+            "const/4 v$sizeCallResultRegister, 0x0".toInstruction()
+        )    
+
+        val (arrayLengthConstIndex, arrayLengthConst) = arrayGenMethodImpl.instructions.withIndex()
+            .first {(it.value as? NarrowLiteralInstruction)?.narrowLiteral == 7 }
+
+        val arrayLengthConstDestination = (arrayLengthConst as OneRegisterInstruction).registerA
+
+        val videoSpeedsArrayType = "Lapp/revanced/integrations/settings/Settings;->videoSpeeds:[F"
```
Comment (TheJeterLP) [inline src/main/kotlin/app/revanced/patches/youtube/misc/customplaybackspeed/patch/CustomPlaybackSpeedPatch.kt:57]: You should use already existing package app.revanced.integrations.videoplayer.videosettings.VideSpeed class for this

**Comment by oSumAtrIX** [comment]
Is this PR ready? If not convert it to a draft.

File: src/main/kotlin/app/revanced/patches/youtube/misc/customplaybackspeed/patch/CustomPlaybackSpeedPatch.kt (hunk 1 above)
Comment (bogadana) [reply src/main/kotlin/app/revanced/patches/youtube/misc/customplaybackspeed/patch/CustomPlaybackSpeedPatch.kt:57]: the was already existing too

Comment (TheJeterLP) [reply src/main/kotlin/app/revanced/patches/youtube/misc/customplaybackspeed/patch/CustomPlaybackSpeedPatch.kt:57]: Yes, but it makes no sense at that place and I want to remove that for cleanup :) Makes no sense to have the stuff at multiple places

Comment (bogadana) [reply src/main/kotlin/app/revanced/patches/youtube/misc/customplaybackspeed/patch/CustomPlaybackSpeedPatch.kt:57]: yeah I'll change it in a moment

**Comment by bogadana** [comment]
Yes it has been ready for days but now it needs https://github.com/revanced/revanced-integrations/pull/22 merged first

**Comment by oSumAtrIX** [comment]
Please check the pending stuff.

**Comment by oSumAtrIX** [comment]
Is the patch usable (without any settings)

**Comment by bogadana** [comment]
yes its hardcoded to these speeds right now {0.25f, 0.5f, 0.75f, 1.0f, 1.25f, 1.5f, 1.75f, 2.0f, 3.0f, 4.0f, 5.0f}

**Comment by oSumAtrIX** [comment]
You are still missing these:
image

**Comment by bogadana** [comment]
is this what you mean?

**Comment by oSumAtrIX** [comment]
Thanks!

**Pull request merged**